lib/tpm2_alg_util.c

```diff
@@ -120,3 +122,123 @@ TPM_ALG_ID tpm2_alg_util_from_optarg(char *optarg) {
     }
     return halg;
 }
+
+bool tpm2_alg_util_is_hash_alg(TPM_ALG_ID id) {
+
+    switch(id) {
+    case TPM_ALG_SHA1:
+        /* fallsthrough */
+    case TPM_ALG_SHA256:
+        /* fallsthrough */
+    case TPM_ALG_SHA384:
+        /* fallsthrough */
+    case TPM_ALG_SHA512:
+        /* fallsthrough */
+    case TPM_ALG_SM3_256:
+        return true;
+        /* no default */
+    }
+
+    return false;
+}
+
+static const char *hex_to_byte_err (int rc) {
+    switch (rc) {
+    case -2:
+        return "String not even in length";
+    case -3:
+        return "Non hex digit found";
+    case -4:
+        return "Hex value too big for digest";
+    }
+    return "unknown";
+}
+
+/* parse digest=<hash|file> digest=<hash|file> ... */
+bool pcr_parse_digest_list(char *digest_list, TPML_DIGEST_VALUES *digests) {
+
+    /* XXX Should we ensure initialization or force the caller? */
+    digests->count = 0;
+
```

Comment:
This is only used to notify the user that no digests were passed right? I would add some flags and set on the 'D' and 'i' case like the other tools do, to make sure that both options were provided or return an error otherwise.